components/SettingsButton.js

```diff
@@ -0,0 +1,45 @@
+import SettingsIcon from "@/public/settings.svg";
+import styled from "styled-components";
+import DeleteButton from "./DeleteButton";
+
+export default function SettingsButton({
+  setEntries,
+  id,
+  toggleModalDeleteButton,
+  handlePopUpSettings,
+  popUpSettings,
+}) {
+  return (
+    <SettingsWrapper>
+      <Button onClick={() => handlePopUpSettings()}>
+        <StyledSettingsIcon />
+      </Button>
```

Comment:
This component and your DeleteButton are essentially the same, genrealize them as described above to use one component for both cases.